sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -270,11 +260,141 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
     keys. Enables resolution of Key Vault references in configuration settings.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, **kwargs) -> None:
         self._dict: Dict[str, str] = {}
         self._trim_prefixes: List[str] = []
         self._client: Optional[AzureAppConfigurationClient] = None
         self._secret_clients: Dict[str, SecretClient] = {}
+        self._key_vault_options: Optional[AzureAppConfigurationKeyVaultOptions] = kwargs.pop("key_vault_options", None)
+        self._selects: List[SettingSelector] = kwargs.pop(
+            "selects", [SettingSelector(key_filter="*", label_filter=EMPTY_LABEL)]
+        )
+        self._sentinel_keys: List[str] = []
+
+        trim_prefixes: List[str] = kwargs.pop("trim_prefixes", [])
+        self._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
+        self._configuration_refresh = self._ConfigurationRefresh(**kwargs)
+
+    def refresh_configuration_settings(self, func):
+        @wraps(func)
+        def refresh_wrapper(*args, **kwargs):
+            self.refresh()
+            return func(*args, **kwargs)
+
+        return refresh_wrapper
+
+    def refresh(self, **kwargs) -> None:
+        # pylint:disable=protected-access
+        refresh_registrations = self._configuration_refresh.refresh_options._refresh_registrations
+        if len(refresh_registrations) == 0:
+            logging.debug("Refresh called but no refresh options set.")
+            self._configuration_refresh.refresh_options._on_error()
+            return
+
+        if datetime.now() < self._configuration_refresh.next_refresh_time:
+            logging.debug("Refresh called but refresh interval not elapsed.")
+            return
+
+        try:
+            # pylint:disable=protected-access
+            for registration in refresh_registrations:
+                if registration.refresh_all:
+                    updated_sentinel = self._client.get_configuration_setting(
+                        key=registration.key_filter, label=registration.label_filter, **kwargs
+                    )
+                    if registration.etag != updated_sentinel.etag:
+                        logging.debug(
+                            "Refresh all triggered by key: %s label %s.",
+                            registration.key_filter,
+                            registration.label_filter,
+                        )
+                        updated_registration = registration
+                        updated_registration_etag = updated_sentinel.etag
+                        self._load_all(**kwargs)
+                        updated_registration.etag = updated_registration_etag
+                        self._configuration_refresh.updated_configurations()
+                        return
+        except Exception as ex:  # pylint:disable=broad-except
```

Comment:
We shouldn't need to catch base Exception anywhere.

Reply:
So, refresh should never throw an exception in my opinion.

Reply:
We can catch the exceptions that we know and care about - any exceptions from AppConfig can be handled here. 
But unexpected exceptions should always be allowed to crash the application because we dont want to hide any serious issues like insufficient memory, app termination, etc.